### DISCUSSION TITLE
Cant get documentation to work setting up the main postal server

### DISCUSSION THREAD

**Discussion by vongohren** [postalserver/postal#3147] [Help with using Postal]
# Intro
Im following these documentations
https://docs.postalserver.io/getting-started/prerequisites
https://docs.postalserver.io/getting-started/installation

Running a VPS with Ubuntu, so it should be quite straight forward. I have set up other systems on the same VPS provider, so it should work.

But the state I end up with does not seem to work very well. I will have a subsection for all questions I have. Hope to 

# Caddy
So just running Caddy with my specific domain fails even if the DNS is properly setup. It fails for the acme challenge calling the .well.known url. I assume the postal server is suppose to be able to answer on request?
What else can be done to make sure this is correctly set up?

# Web server
## Accessing the server 
This is running in docker, but from my understanding I dont see any exposed ports, and question its viability to be accessed?
<img width="1128" alt="image" src="https://github.com/user-attachments/assets/9a7a82de-7de3-4386-9d49-ce44ce6ac568">

## Internals of the server
When attaching to the server I see alot of reference to docker but assume that is not the way it runs, as its a ruby machine and puma is running on the server? Where are the logs for this, so I can verify curling the localhost inside the web server.

**Comment by willpower232** [comment]
> This is running in docker, but from my understanding I dont see any exposed ports, and question its viability to be accessed?

The default setup uses `network_mode: host` which means you don't see the ports in Docker, you can probably see them being used with `sudo netstat -lnp`.

> So just running Caddy with my specific domain fails even if the DNS is properly setup. It fails for the acme challenge calling the .well.known url. I assume the postal server is suppose to be able to answer on request?

Caddy should be able to answer the certificate generation request since that is a feature of Caddy after all, do you definitely have the domain name pointing at your server and no firewalls blocking the web ports?

**Reply by vongohren** [reply]
Yeah DNS is pointing to the IP of the server. But not much is happening. Have another server running n8n and traefik and it all works just fine over there.

Netstat shows everything needed so can see the nessecary ports yes.

Curling the localhost:5000, responds with very little, is it suppose to?

**Reply by willpower232** [reply]
The login form isn't the most amount of HTML to be fair, can you see the `<title>` tag?

**Reply by vongohren** [reply]
No i cannot see any HTML, which makes this a bit tricky. But I do get 403 logs on the server. But I expected to get some html?

**Reply by willpower232** [reply]
Ah Rails does this thing where it only trusts certain hostnames, I think it will only work by the value of web_hostname in your config file (or whatever the equivalent environment variable is)

https://github.com/postalserver/postal/blob/da90e75036c27482699921613d838f4058a100e7/config/application.rb#L43

**Reply by vongohren** [reply]
Ah good to know! Now in atleast returning some HTML with this curl. (just did not provide the real URL)

` curl -H "Host: mypath.mydomain.com" http://127.0.0.1:5000`

This gives me some confidence that postal is running pretty correctly, then I have some issues with exposing further on in my VPS stack, which I presume is out of bounds for the postal documentation

Will try to inform when I get further.

**Reply by vongohren** [reply]
So there were some internal issues with how things was beeing access on my VPS. Discussing with ChatGPT got me through this and I can now access things. I believe it was the ufw settings, but need to verify this with a friend, as my other server, running in same VPS, but with treafik, does not have any ufw settings

I have logged in, added domain, but I have an issue with adding mail server.

```
2024-10-19 19:56:51 +0000 ERROR  POST /org/vongohren/servers (500) event=request transaction=c500753e1cc04bd48071 controller=ServersController action=create format=json method=POST path=/org/vongohren/servers request_id=6d5cb832-c5d3-4b07-b73c-359f3fd21015 ip_address=88.94.15.169 status=500 db_runtime=4.016248941421509 exception_class=NoMethodError exception_message=undefined method `query' for nil:NilClass exception_backtrace=/opt/postal/app/lib/postal/message_db/database.rb:326:in `query_on_connection'\n/opt/postal/app/lib/postal/message_db/database.rb:318:in `block in query'\n/opt/postal/app/lib/postal/message_db/connection_pool.rb:20:in `use'\n/opt/postal/app/lib/postal/message_db/database.rb:345:in `with_mysql'
88.94.15.169 - - [19/Oct/2024:19:56:51 +0000] "GET /500 HTTP/1.1" 500 46 0.0144
```

I dont know what might lead to this, as mariaDB works fine, stores users and domains.

**Reply by willpower232** [reply]
This kind of error usually means that the second set of database configuration is wrong in some way or the user doesn't have access to database names starting with `postal-`.

Hopefully there would be a stack trace after this line in the logs telling you what the problem is.

**Reply by vongohren** [reply]
@willpower232 no unfortunately there is no stack trace in the logs. Or do you mean the logs that you can scroll to the right for above? I get some stacktraces, but did not become any smarter, ref picture. And before I only get a succsesfull request and after the 500 I have no more logs. I cannot find anything relevant in any other postal instances, worker, mariadb or anything

<img width="1170" alt="Screenshot 2024-10-20 at 20 02 36" src="https://github.com/user-attachments/assets/4148f65d-6d17-438d-8c4a-68ef4cb5743a">

**Reply by willpower232** [reply]
Ah I wonder if it goes to a separate log file in the container somewhere.

Did you double check the second set of database configuration? Either way you specify your config, you have to give it the connection details twice for `main_db` and `message_db`

https://github.com/postalserver/postal/blob/da90e75036c27482699921613d838f4058a100e7/doc/config/yaml.yml#L53-L81

**Reply by vongohren** [reply]
@willpower232 haha thanks! Thanks for the follow ups. Any documentation i should update to reflect some of these troubles? This is great and I absolutely did this setup to fast and forgot all about the config file. Forgot to update password for message db. Silly.

Im almost there now, but not receiving messages for some reason. I was able to send to myself and seeing the email in the gmail inbox. Which is great. But I could not send from my gmail to the domain i setup.

Are there any good documentation on debugging this? I looked through the documenation and did not find any good guidance.

My mx records and other DNS setup is up and running and confirmed by the DNS checker in postal.
I could not telnet to the server, on domain 25.

I just added some more of the settings stated here: https://docs.postalserver.io/getting-started/dns-configuration, waiting for propagation. Are they an effect maybe? SPF txt and mx1 and mx2 A records?

**Reply by willpower232** [reply]
>  I could not telnet to the server, on domain 25.

That would be the biggest issue, if your port 25 isn't open then you won't be able to receive emails.

> SPF txt and mx1 and mx2 A records?

You'll need MX records for your domain to be pointing at the Postal server yes.

**Reply by vongohren** [reply]
So things are rolling now, forgot to update the domain itself to the new MX servers.

Anything I should update the documentation with you feel? Or should this all be obvious 😅 

One question. If something is held, and you click deliver. Should it not arrive at incomming messages? Or how is that logic working? Anywhere i can read about this?

**Reply by willpower232** [reply]
Honestly its been so long I couldn't be sure what is obvious and what isn't. If you feel the documentation could be worded better then by all means, submit a PR https://github.com/postalserver/docs

I don't think incoming messages can be held, only outgoing messages. If you've clicked deliver, it probably appears in the same place you found it but with a different status on now.

**Reply by vongohren** [reply]
Ok ok I will evaluate! 

No I was stupid again, had a spam filter on after clicking a button. But it was not spam and others. It was only spam 😅
